include/logger/common.hpp

```diff
@@ -3,24 +3,132 @@
 
 #include <memory>
 #include <string>
+#include <map>
+#include <vector>
 
 namespace logger {
 /**
- * @brief  Log level with their severity order , 0 =>high priority , 5=> least priority.
+ * @brief  Log Levels with their severity order , 0 =>high priority , 5=> least priority.
  */
-enum class LEVEL : int { OFF = 0, FATAL = 1, ERROR = 2, WARN = 3, DEBUG = 4, TRACE = 5 };
+enum class log_level : int { OFF = 0, FATAL = 1, ERROR = 2, WARN = 3, DEBUG = 4, TRACE = 5 };
 
 /**
- * @brief  Interface for the extracted log data like file name , function name etc , one may extend for more
- * parameters such as number of argumnets __VA_ARGS__.
+ *@brief Utility map to store log_levels as key and corresponding value as its equivalent string.
  */
-struct ExtractedLogCredentials
+const std::map<logger::log_level, const std::string> log_level_to_string{ { logger::log_level::OFF, "OFF" },
+  { logger::log_level::FATAL, "FATAL" },
+  { logger::log_level::ERROR, "ERROR" },
+  { logger::log_level::WARN, "WARN" },
+  { logger::log_level::DEBUG, "DEBUG" },
+  { logger::log_level::TRACE, "TRACE" } };
+
+/**
+ * @brief  To deal with extracted log data like file name , function name etc.
+ */
+class ExtractedLogCredentials
 {
-  std::string date = "";
-  std::string time = "";
-  std::string file_name = "";
-  std::string function_name = "";
-  int line_number = 0;
+public:
+  /**
+   * @brief Ctor to initialize various log credentials variables.
+   * @param date  date at which message was logged
+   * @param time  time at which message was logged
+   * @param file_name  file name where message was logged
+   * @param function_name  function name at which message was logged
+   * @param line_number  line number of  the logged message    *
+   * @retval None
+   */
+  ExtractedLogCredentials(const std::string& date = "",
+    const std::string& time = "",
+    const std::string& file_name = "",
+    const std::string& function_name = "",
+    const int& line_number = 0);
+
+  /**
+   * @brief Method to check whether all the data like date, time etc are captured or not.
+   * @retval bool
+   */
+  bool CheckValidLogCredentials();
+
+  /** @defgroup group1 Getters
+   * Getters
+   * @{
+   */
+
+  /**
+   * @brief Returns the date at which message was logged.
+   * @retval None
+   */
+  std::string GetDate() const;
```

Comment:
Do you really required this getter Functiomns. as all Strings are assigned at construction time can we make them const and hence making it public will suffice the need.